core/unit_test/TestViewCtorDimMatch.hpp

```diff
@@ -0,0 +1,429 @@
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+namespace Test {
+
+#define LIVE(EXPR, ARGS, DYNRANK) EXPECT_NO_THROW(EXPR)
+#define DIE(EXPR, ARGS, DYNRANK)                                           \
+  ASSERT_DEATH(                                                            \
+      EXPR,                                                                \
+      "Constructor for Kokkos View 'v_" #ARGS                              \
+      "' has mismatched number of arguments. Number of arguments = " #ARGS \
+      " but dynamic rank = " #DYNRANK)
+
+#define PARAM_0
+#define PARAM_1 1
+#define PARAM_2 1, 1
+#define PARAM_3 1, 1, 1
+#define PARAM_4 1, 1, 1, 1
+#define PARAM_5 1, 1, 1, 1, 1
+#define PARAM_6 1, 1, 1, 1, 1, 1
+#define PARAM_7 1, 1, 1, 1, 1, 1, 1
+
+#define PARAM_0_RANK 0
+#define PARAM_1_RANK 1
+#define PARAM_2_RANK 2
+#define PARAM_3_RANK 3
+#define PARAM_4_RANK 4
+#define PARAM_5_RANK 5
+#define PARAM_6_RANK 6
+#define PARAM_7_RANK 7
+
+using DType = int;
+
+// Skip test execution when KOKKOS_ENABLE_OPENMPTARGET is enabled until
+// Kokkos::abort() aborts properly on that backend
+#if defined(KOKKOS_ENABLE_OPENMPTARGET)
+#else
+TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_params_dyn) {
+  GTEST_FLAG_SET(death_test_style, "threadsafe");
```

Comment:
```
/var/jenkins/workspace/Kokkos/core/unit_test/TestViewCtorDimMatch.hpp: In member function 'virtual void Test::openmp_DeathTest_view_construction_with_wrong_params_dyn_Test::TestBody()':

/var/jenkins/workspace/Kokkos/core/unit_test/TestViewCtorDimMatch.hpp:83:18: error: 'death_test_style' was not declared in this scope

   GTEST_FLAG_SET(death_test_style, "threadsafe");
```

Reply:
As far as I can see we are missing `GTEST_FLAG_SET` completely in our copy of gtest.
Anyways, looking at the implementation the previous version should work just fine:
https://github.com/google/googletest/blob/25ad42aabeaba6227f37795cdd2752e128e83827/googletest/include/gtest/internal/gtest-port.h#L2219

Reply:
This should work. I'll amend this.
`GTEST_FLAG_SET(GTEST_DEFAULT_DEATH_TEST_STYLE, "threadsafe"); `

Reply:
I've rolled back the commit.